### ISSUE TITLE
Add text to image button is shown even when the cover block cannot be inserted

### ISSUE THREAD

**Issue by tomjn** [WordPress/gutenberg#32520]
## Description

There is a transform on the image block that has been turned into a button so that there is an easy shortcut. This button is shown when the cover block is registered.

However, if the cover block settings are filtered so that support for insertion and transforms are removed, the button persists, resulting in broken things ( this site does not support the cover block in these contexts ).

I can find no other solution for removing this ability other that to complete unregister the cover block which is not an option. Users should not be able to insert cover blocks in this context yet this button makes this possible because the `supports` section is ignored.

## WordPress information

- WordPress version: 5.7+

## Device information
- Device: Desktop
- Operating system: MacOS 11
- Browser: Chrome

**Comment by Mamaduka** [comment]
Thanks for reporting this, @tomjn.

Can you share the code for filtering cover block settings?

**Comment by Mamaduka** [comment]
Hi, @tomjn

I've updated some checks recently for this transformation button #32666.

**Comment by tomjn** [comment]
Apologies for not getting back on the code example, I don't have a clean code example, I'd have to extract it, but it can be reproduced by making the cover block not support insertions. We use a filter to turn the `true` to `false` if the block isn't in a whitelist, likewise we disable the transforms so that images can't be transformed into cover blocks ( we remove it from the block definition ), but this button bypasses that

**Comment by tomjn** [comment]
TLDR: the button transforms an image block into a cover block, even when that transformation is removed, and when cover blocks supports section is modified by filters so that it cannot be inserted. The only way to remove the button at the moment is to completely unregister the cover block

The change in #32666 may help mitigate this though

**Comment by Mamaduka** [comment]
Thanks for the feedback, @tomjn.

We probably need `canTransformToBlock` or a similar method that handles various checks for the block. I think this is similar to what @mtias suggested here - https://github.com/WordPress/gutenberg/pull/32666#discussion_r656079988.

**Comment by Mamaduka** [comment]
Hi, @tomjn

Created second PR that should probably cover all cases - #33095.
